File: Java/String-2/src/XyzThere.java
public class XyzThere {
    public static void main(String[] args) {
        /*
         * Return true if the given string contains an appearance of "xyz" where
         * the xyz is not directly preceeded by a period (.). So "xxyz" counts but "x.xyz" does not.
         *
         * xyzThere("abcxyz") → true
         * xyzThere("abc.xyz") → false
         * xyzThere("xyz.abc") → true
         */
        System.out.println(xyzThere("abcxyz"));
        System.out.println(xyzThere("abc.xyz"));
        System.out.println(xyzThere("xyz.abc"));
    }

    public static boolean xyzThere(String str) {
        if (str.startsWith("xyz")) return true;
        for (int i = 0; i < str.length() - 3; i++) {
            if (str.charAt(i) != '.' && str.startsWith("xyz", i + 1)) return true;
        }
        return false;
    }
}

File: Python/String-2/xyz_there.py
def xyz_there(str):
    """
    Return True if the given string contains an appearance of "xyz" where the xyz is not directly
    preceded by a period (.).
    So "xxyz" counts but "x.xyz" does not.

    xyz_there('abcxyz') → True
    xyz_there('abc.xyz') → False
    xyz_there('xyz.abc') → True
    """
    if not "xyz" in str:
        return False

    for i in range(0, len(str) - 2):
        if str[i: i + 3] == "xyz" and (i == 0 or not str[i - 1] == "."):
            return True
    return False


print(xyz_there('abcxyz'))
print(xyz_there('abc.xyz'))
print(xyz_there('xyz.abc'))
